### PR TITLE
Implemented conversion from vec3 to transform

### PR DESCRIPTION
**Pull request by Sheepyhead**
# Objective

Allow conversion from Vec3 to Transform with corresponding translation

## Solution

Implement From<Vec3> for Transform to provide an alternative to using the more wordy 
`Transform::from_translation(Vec3::new(x,y,z))`
This lets you do 
`Vec3::new(x,y,z).into()` 
instead

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_transform/src/components/transform.rs [hunk 1]
```diff
@@ -277,3 +277,9 @@ impl Mul<Vec3> for Transform {
         self.mul_vec3(value)
     }
 }
+
+impl From<Vec3> for Transform {
```
Comment (alice-i-cecile) [inline crates/bevy_transform/src/components/transform.rs:281]: Docs plz. While "into translation" is the obvious conversion, it's not the only possible one.

**Comment by DJMcNab** [comment]
It's a bit strange to use `Transform::from_translation(Vec3::new(xyz))` as the example of current idiomatic code, when that would use `[`Transform::from_xyz`](https://docs.rs/bevy/latest/bevy/prelude/struct.Transform.html#method.from_xyz). I think for cases where you would use the form you gave, of `Vec3::new(x, y, z).into()`, you'd be much better

That being said, this could still be useful in circumstances where you happen to have a `Vec3`; not sure how likely that is in practice. All in all I'm neutral on this change

**Comment by Sheepyhead** [comment]
> It's a bit strange to use `Transform::from_translation(Vec3::new(xyz))` as the example of current idiomatic code, when that would use [`Transform::from_xyz`](https://docs.rs/bevy/latest/bevy/prelude/struct.Transform.html#method.from_xyz). I think for cases where you would use the form you gave, of `Vec3::new(x, y, z).into()`, you'd be much better off using `from_xyz`, as that documents it's a `Transform` being created.
> 
> That being said, this could still be useful in circumstances where you happen to have a `Vec3`; not sure how likely that is in practice. All in all I'm neutral on this change

I speak from personal experience, but generally when I have to create a Transform, 100% of the time I'm extremely concerned with *where* the transform is, and not very concerned about anything else about it. In that regard a Vec3 is interchangable with Transform.translation in my mental model of the game, and having this connection between them makes it a lot easier to reason quickly about it in my opinion. Disagreements are welcome, I'm just explaining my rationale for this. This is inspired by the fact that Rapier does the same thing for their positioning type

**Comment by CptPotato** [comment]
I personally prefer `Transform::from_translation() / ::from_xyz()`.

There isn't a single obvious way to go from a vector to a transform, so `from() / into()` leaves you "guessing" what it'll do. The vector could be a scaling factor, translation, euler rotation, etc; Of course the docs clarify what's happening but I think `From` and `Into` are best used when the conversion is obvious.

**Comment by Sheepyhead** [comment]
@CptPotato Yeah exactly, personally I think the conversion IS obvious but you can definitely make the argument that it isn't. Regardless the PR is here, I leave no judgement about what is decided

**Comment by mockersf** [comment]
Closing as that isn't a big usability improvement and would add some confusion as to which part of the transform would be concerned

**Pull request closed without merging**